src/lib/operators/table_scan.cpp

```diff
@@ -93,9 +93,13 @@ std::shared_ptr<const Table> TableScan::_on_execute() {
 
   for (ChunkID chunk_id{0u}; chunk_id < in_table->chunk_count(); ++chunk_id) {
     if (excluded_chunk_set.count(chunk_id)) continue;
+    const auto& chunk_in = in_table->get_chunk(chunk_id);
+    if (!chunk_in) continue;
 
     auto job_task = std::make_shared<JobTask>([=, &output_mutex, &output_chunks]() {
-      const auto chunk_guard = in_table->get_chunk(chunk_id);
+      const auto& chunk_guard = in_table->get_chunk(chunk_id);
```

Comment:
I don't know if this is possible, but following would be a worst-case scenario: 

1. Chunk available → `job_task` gets created
2. Chunk is physically deleted
3. `job_task` starts with nullpointer.

Therefore, it makes sense to place a chunk-guard

* before the construction of a `job_task` to avoid "empty" jobs
* at the beginning of a `job_task` to ensure null-pointer safety.

Reply:
But isn't this the same as `chunk_in`? Why don't we simply **copy** that into the lambda?

Reply:
Definitely a possibility. In consequnce, we would claim ownership of the chunk in the time between job creation and execution. 
Our MvccDeletePlugin could be blocked by this in terms of physical delete. Otherwise, `TableScan` works with a TransactionContext. So conflicts are prevented anyhow.

Reply:
Not sure if I understand what this finally means. Are you waiting for a response or are you planning to change this. If so, how?

Reply:
Sorry, I just wrote down my thoughts. Nevermind.
I'll remove the check within the `job_task`-lambda. One check is enough. Our MvccDeletePlugin won't cause any conflicts.